service/history/api/respondworkflowtaskcompleted/api.go

```diff
@@ -1042,6 +1050,27 @@ func (handler *WorkflowTaskCompletedHandler) clearStickyTaskQueue(ctx context.Co
 	return nil
 }
 
+func (handler *WorkflowTaskCompletedHandler) validateVersioningInfo(
+	request *workflowservice.RespondWorkflowTaskCompletedRequest,
+	ms workflow.MutableState,
+) error {
+	taskDeployment := worker_versioning.DeploymentFromStamp(request.GetWorkerVersionStamp())
+	wfDeployment := ms.GetCurrentDeployment()
+	if taskDeployment.Equal(wfDeployment) {
```

Comment:
is this supposed to be `!taskDeployment.Equal(wfDeployment)` instead? I'm confused by this case / error

Reply:
yes, good catch.